lasy/profiles/longitudinal/longitudinal_profile_from_data.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+from scipy.constants import c
+
+from .longitudinal_profile import LongitudinalProfile
+
+
+class LongitudinalProfileFromData(LongitudinalProfile):
+    """
+    Derived class for longitudinal laser profile created using
+    data from an experimental measurement or from the output
+    of another code.
+
+    Uses user supplied data to define the longitudinal profile
+    of the laser pulse.
+
+    The data may be supplied in either the spectral or temporal
+    domain. The data should be passed as a structure (defined
+    below). If spectral data is passed, it will be converted to
+    the temporal domain.
+
+    Parameters
+    ----------
+    data : structure
+        The data structure comprises several items indexed by
+        a series of keys
+
+        datatype : string
+            The domain in which the data has been passed. Options
+            are 'spectral' and 'temporal'
+
+        abscissa : ndarrays of floats
+            The horizontal axis of the pulse duration measurement
+            When datatype is 'spectral' abscissa is wavelength in
+            meters
+            When datatype is 'temporal' abscissa is time in seconds
+
+        intensity : ndarrays of floats
+            The vertical axis of the pulse duration measurement
+            When datatype is 'spectral' intensity is spectral
+            intensity. This will be noramlized
+            When datatype is 'temporal' intensity is temporal
+            intensity. This will be normalized
+
+        phase : ndarrays of floats
+            If provided, this phase will be added to the pulse.
+            When datatype is 'spectral' phase is spectral
+            phase.
+            When datatype is 'temporal' phase is temporal
+            phase
```

Comment:
Any reason to cut those lines? <80 is restrictive enough.
```suggestion
            When datatype is 'spectral' phase is spectral phase.
            When datatype is 'temporal' phase is temporal phase.
```